src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -8,31 +8,46 @@ namespace Elastic.Apm.Config
 {
 	public abstract class AbstractConfigurationReader
 	{
-		protected AbstractConfigurationReader(AbstractLogger logger) => Logger = logger;
+		protected AbstractConfigurationReader(IApmLogger logger) => ScopedLogger = logger?.Scoped(GetType().Name);
 
-		protected AbstractLogger Logger { get; }
+		private ScopedLogger ScopedLogger { get; }
+
+		protected IApmLogger Logger => ScopedLogger;
 
 		protected static ConfigurationKeyValue Kv(string key, string value, string origin) =>
 			new ConfigurationKeyValue(key, value, origin);
 
-		protected internal static LogLevel ParseLogLevel(string value) =>
-			Enum.TryParse<LogLevel>(value, out var logLevel) ? logLevel : AbstractLogger.LogLevelDefault;
+
+		protected internal static LogLevel ParseLogLevel(string value)
+		{
+			switch (value.ToLowerInvariant())
+			{
+				case "trace": return LogLevel.Trace;
+				case "debug": return LogLevel.Debug;
+				case "information":
+				case "info": return LogLevel.Information;
```

Comment:
Could you clarify please why do we need both `information` and `info` - wouldn't `info` be enough (and naming the enum `LogLevel.Info`)?

Reply:
I basically mirrored the enum from `Microsoft.Extensions.Logging` which lists `Information`. As a config option supporting both made sense to me